src/simulators/statevector/qubitvector_thrust.hpp

```diff
@@ -369,69 +1106,64 @@ class QubitVectorThrust {
   //-----------------------------------------------------------------------
   // Statevector update with Lambda function
   //-----------------------------------------------------------------------
-  	template <typename UnaryFunction>
-	void apply_function(UnaryFunction func,const reg_t &qubits);
+  template <typename Function>
+  double apply_function(Function func,const reg_t &qubits) const;
```

Comment:
Can you write a short usage description for these functions, im not sure what they do:
* `apply_function`
* `set_matrix`
* `set_params`
* `allocated_buffers`